src/transformers/trainer.py

```diff
@@ -1381,6 +1404,32 @@ def _maybe_log_save_evaluate(self, tr_loss, model, trial, epoch):
             self._save_checkpoint(model, trial, metrics=metrics)
             self.control = self.callback_handler.on_save(self.args, self.state, self.control)
 
+    def _load_rng_state(self, checkpoint):
```

Comment:
Could this method also print a warning in case we're on TPU as we don't expect reproducibility when on TPUs?

Reply:
Mmm, maybe just a comment in the README where we document resuming from checkpoint? I don't really want to issue a warning for each run on TPU using a checkpoint.